packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -53,6 +54,15 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
       console.warn(`[${componentName}] \`${main}\` and \`${sub}\` cannot be used together. \`${sub}\` will be ignored.`)
     }
   }
+  const resolveTrailingSlashBehavior = (to: RouteLocation): RouteLocationPathRaw => {
+    const trailingSlash = options.trailingSlashBehavior === 'append' ? '/' : ''
+    // modify trailing slash in "path", exclude "name" to apply trailing slash behavior
+    return {
+      path: to.path.replace(/\/$/, '') + trailingSlash,
+      query: to.query,
+      hash: to.hash
+    }
+  }
```

Comment:
I think we can improve this a little:

1. if route passed is a string, we can use `withTrailingSlash` from `ufo` to add the slash and return the modified string
2. if route passed is an object without path, we can call `router.resolve` on the object to fully resolve it.
3. once we have path, we return an object with `.path` and `.fullPath` values with trailing slashes added.

Reply:
Ok to all, except that `fullPath` is set by `vue-router`, when it resolves privided `to/href`, so I think only `path` should be modified.